snapcraft/internal/elf.py

```diff
@@ -153,15 +156,20 @@ def is_linker_compatible(self, *, linker: str) -> bool:
 
     def get_required_glibc(self) -> str:
         """Returns the required glibc version for this ELF file."""
+        with contextlib.suppress(AttributeError):
+            return self._required_glibc  # type: ignore
```

Comment:
I would prefer to see the more familiar pattern of setting this to `None` in `__init__` and setting it if so before returning here (like we do elsewhere), but I don't have strong feelings about it.

Reply:
I am divided, now that `None` and `''` means you are changing types :confused:

Reply:
Nah, `self._required_glibc = None # type: str`. You could use `self._required_glibc = ''` and get pretty much the same thing, too.